mysk-data-api/src/routes/v1/subjects/electives/trade_offers/update_offer.rs

```diff
@@ -0,0 +1,168 @@
+use crate::{
+    extractors::{api_key::ApiKeyHeader, student::LoggedInStudent},
+    AppState,
+};
+use actix_web::{
+    put,
+    web::{Data, Json, Path},
+    HttpResponse, Responder,
+};
+use mysk_lib::{
+    common::{
+        requests::{QueryablePlaceholder, RequestType, SortablePlaceholder},
+        response::ResponseType,
+    },
+    models::{
+        elective_trade_offer::{db::DbElectiveTradeOffer, ElectiveTradeOffer},
+        enums::SubmissionStatus,
+        traits::TopLevelGetById as _,
+    },
+    prelude::*,
+};
+use serde::Deserialize;
+use sqlx::{query, query_as};
+use uuid::Uuid;
+
+#[derive(Debug, Deserialize)]
+struct UpdatableElectiveOffer {
+    pub status: SubmissionStatus,
+}
+
+#[allow(clippy::too_many_lines)]
+#[put("/{id}")]
+async fn update_trade_offer(
+    data: Data<AppState>,
+    trade_offer_id: Path<Uuid>,
+    request_body: Json<
+        RequestType<UpdatableElectiveOffer, QueryablePlaceholder, SortablePlaceholder>,
+    >,
+    student_id: LoggedInStudent,
+    _: ApiKeyHeader,
+) -> Result<impl Responder> {
+    let pool = &data.db;
+    let client_student_id = student_id.0;
+    let trade_offer_id = trade_offer_id.into_inner();
+    let trade_offer_status = match &request_body.data {
+        Some(request_data) => match request_data.status {
+            SubmissionStatus::Approved | SubmissionStatus::Declined => request_data.status,
+            SubmissionStatus::Pending => {
+                return Err(Error::InvalidRequest(
+                    "Status must be either 'approved' or 'declined'".to_string(),
+                    format!("/subjects/electives/trade-offers/{trade_offer_id}"),
+                ));
+            }
+        },
+        _ => unreachable!("JSON errors are pre-handled by the JsonConfig error handler"),
+    };
+    let fetch_level = request_body.fetch_level.as_ref();
+    let descendant_fetch_level = request_body.descendant_fetch_level.as_ref();
+
+    let Some(trade_offer) = query_as!(
+        DbElectiveTradeOffer,
+        r#"
+        SELECT
+            id,
+            created_at,
+            sender_id,
+            receiver_id,
+            status AS "status: SubmissionStatus",
+            sender_elective_subject_id,
+            receiver_elective_subject_id
+        FROM elective_subject_trade_offers
+        WHERE id = $1 AND status = $2
+        "#,
+        trade_offer_id,
+        SubmissionStatus::Pending as SubmissionStatus,
+    )
+    .fetch_optional(pool)
+    .await?
+    else {
+        return Err(Error::EntityNotFound(
+            "Trade offer not found".to_string(),
+            format!("/subjects/electives/trade-offers/{trade_offer_id}"),
+        ));
+    };
+
+    let mut updated_status: Option<SubmissionStatus> = None;
+
+    if client_student_id == trade_offer.sender_id {
```

Comment:
Why are there no check for if client_student_id is neither a sender or a receiver